src/main/java/org/sitenv/vocabularies/validation/validators/nodetypes/RequiredNodeValidator.java

```diff
@@ -0,0 +1,98 @@
+import org.sitenv.vocabularies.configuration.ConfiguredValidator;
+import org.sitenv.vocabularies.validation.dto.NodeValidationResult;
+import org.sitenv.vocabularies.validation.dto.VocabularyValidationResult;
+import org.sitenv.vocabularies.validation.dto.enums.VocabularyValidationResultLevel;
+import org.sitenv.vocabularies.validation.utils.ConfiguredExpressionFilter;
+import org.sitenv.vocabularies.validation.utils.XpathUtils;
+import org.sitenv.vocabularies.validation.validators.NodeValidator;
+import org.springframework.stereotype.Component;
+
+import com.ximpleware.VTDNav;
+
+@Component(value = "RequiredNodeValidator")
+public class RequiredNodeValidator extends NodeValidator {
+	
+	private static Logger LOGGER = Logger.getLogger(RequiredNodeValidator.class); 
+	private static final String AT_SIGN = "@";
+	private static final short ONE = 1;
+
+	@Override
+	public List<VocabularyValidationResult> validateNode(ConfiguredValidator configuredValidator, XPath xpath,
+			VTDNav vn, int nodeIndex, ConfiguredExpressionFilter filter, String xpathExpression) {	
+		boolean hasNode = false;
+
+    	LOGGER.debug("Entered RequiredNodeValidator.validateNode() with xpathExpression = " + xpathExpression);
+    	
+        try {
+        	String nodeName = configuredValidator.getRequiredNodeName();
+//        	LOGGER.debug("nodeName: " + nodeName);
+        	
+        	if (nodeName.startsWith(AT_SIGN)) { // It's an attribute such as @unit
+        		String attrName = nodeName.substring(ONE);
+        		hasNode = vn.hasAttr(attrName);
+//        		LOGGER.debug("Found Attribute: " + attrName);
+        	} else if (StringUtils.isNotBlank(nodeName)) { // It's an element
+        		hasNode = vn.matchElement(nodeName);
+//        		LOGGER.debug("Found Element: " + nodeName);        		
+        	} else {
+        		LOGGER.error("Found null or blank nodeName: " + nodeName); 
+        	}
+        	
+//        	LOGGER.debug("hasNode = " + hasNode);
+        } catch (Exception e) {
+			throw new RuntimeException("ERROR parsing document with given XPath expression: " + e.getMessage());
+        }
+
+		NodeValidationResult nodeValidationResult = new NodeValidationResult();
+        nodeValidationResult.setErrorOffset(vn.getTokenOffset(vn.getCurrentIndex()));
+        nodeValidationResult.setValidatedDocumentXpathExpression(xpathExpression);
+        nodeValidationResult.setRequestedNode(configuredValidator.getRequiredNodeName());
+        nodeValidationResult.setConfiguredValidationMessage(configuredValidator.getValidationMessage());
+        nodeValidationResult.setRuleID(configuredValidator.getId());
+        
+        if(hasNode) {
+        	nodeValidationResult.setValid(true);
+        }
+        
+		return buildVocabularyValidationResults(nodeValidationResult, configuredValidator.getConfiguredValidationResultSeverityLevel(), filter, vn);    
+	}
+
+	@Override
+	protected List<VocabularyValidationResult> buildVocabularyValidationResults(
```

Comment:
Can you have another overloaded method and call this ?

Reply:
I can, but it doesn't make sense to do it.  If you check the peer classes extending from the NodeValidator, inside their implementation of the buildVocabularyValidationResults(...) method, some use the filter - such as ValueSetCodeValidator class, many don't.  The ones who don't (such as the TextNodeValidator) don't use an overloaded method to call the overridden method.  So it's a toss, I don't see the gain to create another layer of method in the call stack just to show 1 input param is not used.

Reply:
Did not notice the "@Override".